vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -5,51 +5,21 @@ class SecureRoomsApi::ScansController < ApplicationController
     password: Settings.secure_rooms_api.basic_auth_password,
   )
 
-  rescue_from ActiveRecord::RecordNotFound, with: :report_missing_ids
-
-  before_action :load_models
+  before_action :load_authorizer
 
   def scan
-    user_accounts = @user.accounts_for_product(@card_reader.secure_room)
-
-    if user_accounts.present?
-      response_status = user_accounts.many? ? :multiple_choices : :ok
-
-      # TODO: needs a legitimage tablet_identifier once tablet exists
-      response_json = {
-        response: "select_account",
-        tablet_identifier: "abc123",
-        name: @user.full_name,
-        accounts: SecureRooms::AccountPresenter.wrap(user_accounts),
-      }
-    else
-      response_status = :forbidden
-      response_json = {
-        response: "deny",
-        reason: "No accounts found",
-      }
-    end
-
-    render json: response_json, status: response_status
-  end
-
-  def load_models
-    @user = User.find_by!(card_number: params[:card_number])
-    @card_reader = SecureRooms::CardReader.find_by!(
-      card_reader_number: params[:reader_identifier],
-      control_device_number: params[:controller_identifier],
-    )
+    @scan_authorizer.perform
+    render @scan_authorizer.response
   end
 
   private
 
-  def report_missing_ids(error)
-    response_json = {
-      response: "deny",
-      reason: error.message,
-    }
-
-    render json: response_json, status: :not_found
+  def load_authorizer
+    @scan_authorizer = SecureRooms::ScanAuthorizer.new(
```

Comment:
My general initial feeling is that we're now packing too much into the authorizer. I think I'd much rather see the authorizer take a User and a Product and return a grant/deny info. Then the controller is responsible for building the JSON response. Maybe there's another object in there that does the translating from the Authorizer to responses.